### PR TITLE
Minimal changes for Material with AppCompat

### PR DESCRIPTION
**Pull request by philipwhiuk**
This is minimal change PR for moving to using Material via the AppCompat library.

There's lots more we can do here - but this is both usable and a step in the right direction.

Main changes:
* Using v7 ActionBar
* K9ListActivity now implements the List functionality and extends K9Activity.
* A few action bar actions are now 'ifRoom' instead of 'always so that there's space for the account name on the Folder list

Screenshots:
https://photos.app.goo.gl/0NoJuo75jkh5U9Tw2

### CODE REVIEW

File: k9mail/src/main/java/com/fsck/k9/activity/Accounts.java [hunk 1]
```diff
@@ -132,7 +132,7 @@
     private FontSizes mFontSizes = K9.getFontSizes();
 
     private MenuItem mRefreshMenuItem;
-    private ActionBar mActionBar;
+    private android.support.v7.app.ActionBar mActionBar;
```
Comment (cketti) [inline k9mail/src/main/java/com/fsck/k9/activity/Accounts.java]: Can you remove the old import for `ActionBar` so the unqualified name can be used here?

File: k9mail/src/main/java/com/fsck/k9/activity/K9ListActivity.java [hunk 2]
```diff
@@ -1,20 +1,22 @@
...
 import com.fsck.k9.K9;
 import com.fsck.k9.activity.K9ActivityCommon.K9ActivityMagic;
 import com.fsck.k9.activity.misc.SwipeGestureDetector.OnSwipeGestureListener;
 
 
-public abstract class K9ListActivity extends ListActivity implements K9ActivityMagic {
+public abstract class K9ListActivity extends K9Activity implements K9ActivityMagic {
 
     private K9ActivityCommon mBase;
+    protected ListAdapter mAdapter;
```
Comment (cketti) [inline k9mail/src/main/java/com/fsck/k9/activity/K9ListActivity.java]: Please don't create new fields with the `m` prefix. I'd rather mix than introduce more of these.

File: k9mail/src/main/java/com/fsck/k9/activity/FolderList.java [hunk 3]
```diff
@@ -95,7 +94,7 @@
 
     private MenuItem mRefreshMenuItem;
     private View mActionBarProgressView;
-    private ActionBar mActionBar;
+    private android.support.v7.app.ActionBar mActionBar;
```
Comment (cketti) [inline k9mail/src/main/java/com/fsck/k9/activity/FolderList.java]: Use unqualified name.

File: k9mail/src/main/java/com/fsck/k9/activity/MessageList.java [hunk 4]
```diff
@@ -149,7 +149,7 @@ public static Intent actionDisplayMessageIntent(Context context,
 
     private StorageManager.StorageListener mStorageListener = new StorageListenerImplementation();
 
-    private ActionBar mActionBar;
+    private android.support.v7.app.ActionBar mActionBar;
```
Comment (cketti) [inline k9mail/src/main/java/com/fsck/k9/activity/MessageList.java]: Use unqualified name.

File: k9mail/src/main/java/com/fsck/k9/activity/K9ListActivity.java (hunk 2 above)
Comment (philipwhiuk) [reply k9mail/src/main/java/com/fsck/k9/activity/K9ListActivity.java]: I'll submit a review to get rid of them and rebase.

**Comment by philipwhiuk** [comment]
Thanks for the feedback. :+1:

**Comment by philipwhiuk** [comment]
Rebased and addressed code review feedback

```
$ grep -rni "android.app.ActionBar" k9mail/src/ | wc -l
0
```

**Comment by cketti** [comment]
The namespace for the `showAsAction` attributes in `message_list_option.xml` need to be updated, too.

I want to hold off on merging this because `master` already contains a lot of good fixes that I don't want to backport to 5.4x. So I want to push out 5.600 rather soon. While I think this is a great start towards material design I don't think it's an intermediate state we should release in a stable version.

**Comment by philipwhiuk** [comment]
I mean the whole point of master is for this kind of patch. If you want to do a beta before this lands then let's cut a branch for 5.500 now and merge this to master.

We can release that stuff now and get a stable out considerably before the runtime permission cut off date. Then this and the runtime permissions can go in 5.700

**Comment by philipwhiuk** [comment]
There were a few other few other files too. I've changed them all.

**Comment by cketti** [comment]
Creating another branch still requires backporting fixes. I'm not talking "don't merge for another 5 months". I want to release a new stable version consisting mostly of bug fixes in February. The things I want to do before a new release are:
* HTML converter rewrite (mostly done)
* attempting to fix the drafts sync issue
* looking into the remaining reported HTML display issues -> extending the whitelist
* looking into issues around encryption

**Comment by fat-tire** [comment]
I haven't tried this but is there a reason you left in some of the holo bits?  Shouldn't it be including more like this stuff?  (note this is untested... I did build a material version for my chromebook, but just used new theme/styles xml in values-v25 then moved it back to use compatlib)

```
diff --git a/k9mail/src/main/res/values/styles.xml b/k9mail/src/main/res/values/styles.xml
index f21f428b1..aac9aa542 100644
--- a/k9mail/src/main/res/values/styles.xml
+++ b/k9mail/src/main/res/values/styles.xml
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="utf-8"?>
 <resources xmlns:tools="http://schemas.android.com/tools">
 
-    <style name="Theme.K9Dialog" parent="@android:Theme.Light">
+    <style name="Theme.K9Dialog" parent="Theme.AppCompat.Light">
         <item name="android:windowFrame">@null</item>
         <item name="android:windowBackground">@drawable/popup_background</item>
         <item name="android:windowTitleStyle">?android:attr/textAppearanceLargeInverse</item>
diff --git a/k9mail/src/main/res/values/themes.xml b/k9mail/src/main/res/values/themes.xml
index e866f3a8b..924d09725 100644
--- a/k9mail/src/main/res/values/themes.xml
+++ b/k9mail/src/main/res/values/themes.xml
@@ -1,13 +1,13 @@
 <?xml version="1.0" encoding="utf-8"?>
 <resources xmlns:android="http://schemas.android.com/apk/res/android">
 
-    <style name="Theme.K9.Startup" parent="@android:style/Theme.Holo.NoActionBar">
+    <style name="Theme.K9.Startup" parent="Theme.AppCompat.Light.NoActionBar">
         <item name="android:windowBackground">@android:color/black</item>
     </style>
 
     <!-- Empty base themes that can be easily replaced by RRO (Runtime Resource Overlay) themes -->
-    <style name="Theme.K9.Light.Base" parent="@android:style/Theme.Holo.Light" />
-    <style name="Theme.K9.Dark.Base" parent="@android:style/Theme.Holo" />
+    <style name="Theme.K9.Light.Base" parent="Theme.AppCompat.Light" />
+    <style name="Theme.K9.Dark.Base" parent="Theme.AppCompat" />
 
     <style name="Theme.K9.Light.Common" parent="Theme.K9.Light.Base">
         <item name="iconFolder">@drawable/folder_light</item>
@@ -152,7 +152,7 @@
         <item name="backgroundColorChooseAccountHeader">#404040</item>
     </style>
 
-    <style name="Theme.K9.Dialog.Translucent.Dark" parent="@android:style/Theme.Holo.Dialog">
+    <style name="Theme.K9.Dialog.Translucent.Dark" parent="Theme.AppCompat.Dialog">
         <item name="android:windowBackground">@android:color/transparent</item>
         <item name="android:colorBackgroundCacheHint">@null</item>
         <item name="android:windowIsTranslucent">true</item>
@@ -165,7 +165,7 @@
         <item name="tintColorBulletPointNeutral">#bbb</item>
     </style>
 
-    <style name="Theme.K9.Dialog.Translucent.Light" parent="@android:style/Theme.Holo.Light.Dialog">
+    <style name="Theme.K9.Dialog.Translucent.Light" parent="Theme.AppCompat.Light.Dialog">
         <item name="android:windowBackground">@android:color/transparent</item>
         <item name="android:colorBackgroundCacheHint">@null</item>
         <item name="android:windowIsTranslucent">true</item>

```

**Comment by philipwhiuk** [comment]
If I missed the Dialogs that's an error. I'll fix that.

**Comment by fat-tire** [comment]
Hey @philipwhiuk -- I got a little carried away with the stuff I was doing for my Chromebook and kept going with this.  I won't submit it officially because yours is already here, but for reference if you want to incorporate any of it, please feel free.  It's basically what you were doing, just w/a few more XML files and the dialogs and a few other things-- it's meant to go over my Android Studio 3.1 commit #3159.  I'll add a note there too in case anyone wants it.

[Material with AppCompat (building on commit from @philipwhiuk)](https://github.com/fat-tire/k-9/commit/d7a5f01ec2c56824190116f08d3cdef2bda39ab1)

**Comment by philipwhiuk** [comment]
Closing in favour of PR from K-9 repository.

**Pull request closed without merging**